ambry-network/src/main/java/com.github.ambry.network/Selector.java

```diff
@@ -347,6 +358,24 @@ public void poll(long timeoutMs, List<NetworkSend> sends)
           close(key);
         }
       }
+
+      Iterator<Transmission> sslTransIter = pendingSslHandshakes.iterator();
+      while (sslTransIter.hasNext()) {
+        Transmission sslTransmission = sslTransIter.next();
+        if (sslTransmission != null && sslTransmission.ready()) {
+          connected.add(sslTransmission.getConnectionId());
+          this.metrics.selectorConnectionCreated.inc();
+          Long handshakeStartTime = sslHandshakeTimer.remove(sslTransmission.getConnectionId());
+          if (handshakeStartTime != null) {
+            this.metrics.selectorPerceivedSslHandshakeTime
+                .update(System.currentTimeMillis() - handshakeStartTime.longValue());
+          }
+          sslTransIter.remove();
+        } else if (sslTransmission == null || !keyMap.containsKey(sslTransmission.getConnectionId())) {
```

Comment:
1. When could the second half be `true`?
2. Should there be a timeout mechanism for ssl handshaking? I see you already have an `sslHandshakeTimer`, which is now being used for calculating time.
3. In any case if the handshake process gets wrong, where should handle this? I will come to you for this question.